### PR TITLE
lis2dw12: add Single/Double Tap triggers

### PR DESCRIPTION
**Pull request by avisconti**
Add support for Single/Double Tap event recognition and trigger
generation. For more information read STM AN5038, section 5.6:

https://www.st.com/resource/en/application_note/dm00401877.pdf

TESTED on IKS01A3 shield on top of Nucleo F401RE board

### CODE REVIEW

**Review by erwango — APPROVED** [review]

**Comment by avisconti** [comment]
I think that is better to wait until #14111 will be accepted and merged.
After that I will re-base this work on top of it.

**Comment by MaureenHelm** [comment]
Do you still want to get this into the 2.0.0 release? Please rebase

**Comment by avisconti** [comment]
> Do you still want to get this into the 2.0.0 release? Please rebase

@MaureenHelm 
It will not be just a rebase, as the lis2dw12 driver has been recently ported by me on top of stmemsc HAL. So, I would need to rewrite entirely this commit. For the moment I'm not interested to have this feature merged and I'm closing the PR right now. In the future I may come back with another PR on this subject. Thanks!

**Pull request closed without merging**
